pkg/cloud/services/compute/instance.go

```diff
@@ -614,13 +623,15 @@ func (s *Service) DeleteInstance(object runtime.Object, instanceName string) err
 			}
 			if len(trunkInfo) == 1 {
 				err = util.PollImmediate(RetryIntervalTrunkDelete, TimeoutTrunkDelete, func() (bool, error) {
+					mc := metrics.NewMetricPrometheusContext("trunk", "delete")
 					if err := trunks.Delete(s.networkClient, trunkInfo[0].ID).ExtractErr(); err != nil {
```

Comment:
I would propose putting `err = mc.ObserveRequest(err)` before the if, so we don't have to do it twice, wdyt?

(would also apply to some other occurrences below)